docs/sources/flow/tasks/migrate/from-otelcol.md

```diff
@@ -0,0 +1,276 @@
+   Do not use this flag in a production environment.
+   {{< /admonition >}}
+
+## Example
+
+This example demonstrates converting a OpenTelemetry Collector configuration file to a {{< param "PRODUCT_NAME" >}} configuration file.
+
+The following OpenTelemetry Collector configuration file provides the input for the conversion.
+
+```yaml
+receivers:
+  otlp:
+    protocols:
+      grpc:
+      http:
+
+exporters:
+  otlp:
+    endpoint: database:4317
+
+processors:
+  memory_limiter:
+    limit_percentage: 90
+    check_interval: 1s
+
+
+service:
+  pipelines:
+    metrics:
+      receivers: [otlp]
+      processors: [memory_limiter]
+      exporters: [otlp]
+    logs:
+      receivers: [otlp]
+      processors: [memory_limiter]
+      exporters: [otlp]
+    traces:
+      receivers: [otlp]
+      processors: [memory_limiter]
+      exporters: [otlp]
+```
+
+The convert command takes the YAML file as input and outputs a [River][] file.
+
+{{< code >}}
+
+```static-binary
+AGENT_MODE=flow grafana-agent convert --source-format=otelcol --output=<OUTPUT_CONFIG_PATH> <INPUT_CONFIG_PATH>
+```
+
+```flow-binary
+grafana-agent-flow convert --source-format=otelcol --output=<OUTPUT_CONFIG_PATH> <INPUT_CONFIG_PATH>
+```
+
+{{< /code >}}
+
+Replace the following:
+
+- _`<INPUT_CONFIG_PATH>`_: The full path to the OpenTelemetry Collector configuration.
+- _`<OUTPUT_CONFIG_PATH>`_: The full path to output the {{< param "PRODUCT_NAME" >}} configuration.
+
+The new {{< param "PRODUCT_NAME" >}} configuration file looks like this:
+
+```river
+otelcol.receiver.otlp "default" {
+	grpc { }
+
+	http { }
+
+	output {
+		metrics = [otelcol.processor.memory_limiter.default.input]
+		logs    = [otelcol.processor.memory_limiter.default.input]
+		traces  = [otelcol.processor.memory_limiter.default.input]
+	}
+}
+
+otelcol.processor.memory_limiter "default" {
+	check_interval   = "1s"
+	limit_percentage = 90
+
+	output {
+		metrics = [otelcol.exporter.otlp.default.input]
+		logs    = [otelcol.exporter.otlp.default.input]
+		traces  = [otelcol.exporter.otlp.default.input]
+	}
+}
+
+otelcol.exporter.otlp "default" {
+	client {
+		endpoint = "database:4317"
+	}
+}
+```
+
+## Limitations
+
+Configuration conversion is done on a best-effort basis. {{< param "PRODUCT_ROOT_NAME" >}} will issue warnings or errors where the conversion can't be performed.
+
+After the configuration is converted, review the {{< param "PRODUCT_NAME" >}} configuration file created and verify that it's correct before starting to use it in a production environment.
+
+The following list is specific to the convert command and not {{< param "PRODUCT_NAME" >}}:
+
+* Many OpenTelemetry Collector components are supported. You can get a general idea of which exist in Flow mode for conversion by reviewing
+  the `otelcol.*` components in the [Component Reference]({{< relref "../../reference/components/" >}}). Any additional unsupported features are returned as errors during conversion.
```

Comment:
Should we make it clearer that we don't support conversions for components that we do not embed directly?

Reply:
I added some words about this.